src/YoutubeAtom.tsx

```diff
@@ -274,26 +318,40 @@ export const YoutubeAtom = ({
         false,
     );
 
+    // to avoid rerenders we use an object outside of React compont's scope
+    // to make sure data is unique to this video and not overriding other youtube
+    // video's data, we key all the data based on videoMeta.assetId
+    if (!sharedYTData[videoMeta.assetId]) {
+        sharedYTData = {
+            ...sharedYTData,
+            [videoMeta.assetId]: initSharedYTData,
+        };
+    }
```

Comment:
Why do we need to use global state here instead of component level state? I'm not at the coal face for trying to get this to work so I get that YT might be doing weird stuff, but my instinct is that the the correct solution would be to do this in a more 'reacty' way.